zio-http/src/main/scala/zio/http/ZClient.scala

```diff
@@ -28,60 +28,43 @@ import zio.http.socket.SocketApp
 import java.net.{InetSocketAddress, URI} // scalafix:ok;
 
 trait ZClient[-Env, -In, +Err, +Out] extends HeaderOps[ZClient[Env, In, Err, Out]] { self =>
```

Comment:
Now that I moved config to `ZClient.Config`, I was wondering it would look better to have `Client` as the "main" companion object, hand have the layers and config class in that instead of `ZClient`. Would be more symmetric to `Server`.

Reply:
How about `val Client = ZClient` or is that too ZIO 1.0-ish?

Reply:
I think that's the best we came up with when using this more polymorphic variant pattern.

Reply:
Don't we just need a type alias? `type Client = ZClient.type`

Reply:
We could do `trait Client[...] extends Client.Poly[...]`